BraintreeCore/src/main/java/com/braintreepayments/api/core/BraintreeClient.kt

```diff
@@ -233,6 +228,19 @@ class BraintreeClient @VisibleForTesting internal constructor(
         }
     }
 
+    @Suppress("TooGenericExceptionCaught")
+    @WorkerThread
+    private fun sendHttpRequestSync(
+        request: InternalHttpRequest,
+        configuration: Configuration
+    ): Pair<String?, Exception?> = try {
```

Comment:
Instead of returning a `Pair<String?, Exception?>`, what are your thoughts in returning some sort of `Result` sealed class? This way we can better handle the result with a when statement instead of having to check for null?

It would also remove 2 of the possible cases that should never happen - when both String and Exception are not null and when they are both null.

Reply:
I 100% agree the `try/catch` has a big time Java feel. It would move us closer towards the coroutine direction as well.

I do have two follow up PRs in mind, one to remove `HttpClient` because the threading logic is no longer needed, and the other is to implement the retry logic in `ConfigurationLoader`. I wonder if this would make sense in the `HttpClient` clean up PR? Or would you prefer we go for it here? I'm cool with either approach.

Reply:
I'm good either way, whichever is easier on your end!

Reply:
Ok cool yeah. I can clean this up in the next one! It will make a clean diff. 🙏